specs/phase1/beacon-chain.md

```diff
@@ -164,6 +163,8 @@ class AttestationData(Container):
     # FFG vote
     source: Checkpoint
     target: Checkpoint
+    # Shard vote
+    shard: Shard
```

Comment:
I think if we will want to make sure that the `shard` and `committee_index` are coherent for on-time attestations via a call to `compute_shard_from_committee_index` in `process_attestation`

Then in `process_crosslinks` and `process_crosslink_for_shard` we can directly use `attestation.data.shard`

(I think we'll force a stub for the late attestations for both shard and committee index (e.g. `2**64 - 1`))

Reply:
This should also be part of p2p validation.  If the attestation is on-time, then the shard and committee_index must be coherent, or else ignore/reject the attestation

Reply:
Whoops, an assertion was lost in the previous auto-merge. Nice catch! 🙏 

> (I think we'll force a stub for the late attestations for both shard and committee index (e.g. 2**64 - 1))

> or else ignore/reject the attestation

Doesn't the late attestation still have to be re-broadcasted to subnet and then get aggregated? I guess we still need the corresponding `commitee_index` field here to locate the correct subnet and compute beacon committee?

Reply:
Ah, you're right. I was over-optimizing what we could potentially aggregate here. 

Will consider this a bit more but the late attestation being bound to the committee and size of the committee is probably reasonable to keep in place